File: backend/DTOs/GetCourseLessonByContentFullTextSearchResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using EduConnect.Enums;

namespace EduConnect.DTOs
{
    public class GetCourseLessonByContentFullTextSearchResponse
    {
        public Guid CourseLessonId { get; set; }
        public Guid CourseLessonContentId { get; set; }
        public string Title { get; set; }
        public string Topic { get; set; }
        public string ShortSummary { get; set; }
        public PublishedStatus PublishedStatus { get; set; }

        public int? LessonSequenceOrder { get; set; }
        public DateTime CreatedAt { get; set; }

        public DateTime? StatusChangedAt { get; set; }

        public string Content { get; set; }



    }
}

File: frontend/EduConnect/src/app/models/course/course-tutor-controller/get-course-lesson-by-content-full-text-search-response.ts
import { PublishedStatus } from '../../../../enums/published-status.enum';

export interface GetCourseLessonByContentFullTextSearchResponse {
  courseLessonId: string;
  courseLessonContentId: string;
  title: string;
  topic: string;
  shortSummary: string;
  publishedStatus: PublishedStatus;
  lessonSequenceOrder: number | null;
  createdAt: string;
  statusChangedAt: string | null;
  content: string;
}
